planner/cascades/testdata/transformation_rules_suite_out.json

```diff
@@ -253,10 +253,27 @@
         "SQL": "select a from (select floor(a) as a from t) as t2",
         "Result": [
           "Group#0 Schema:[Column#13]",
+          "    Projection_4 input:[Group#1], floor(test.t.a)->Column#13",
           "    Projection_2 input:[Group#1], floor(test.t.a)->Column#13",
           "Group#1 Schema:[test.t.a]",
           "    TableScan_1 table:t"
```

Comment:
Not related to this PR.
These two Projections are actually the same. We need a method to merge these two GroupExpr now. cc @winoros 
One solution is to implement `HashCode()` for GroupExpr, then we only need to check the HashCode before we insert a new GroupExpr into the Group.

Reply:
Oh, this is not a MergeGroup case, because they have the same childGroup. It only needs a deduplication.